mis_builder_total_committed_purchase/data/mis_total_committed_purchase.sql

```diff
@@ -0,0 +1,67 @@
+CREATE OR REPLACE VIEW mis_total_committed_purchase AS (
+    SELECT ROW_NUMBER() OVER() AS id, mis_total_committed_purchase.* FROM (
+        WITH currency_rate as (
+            SELECT
+              r.currency_id,
+              COALESCE(r.company_id, c.id) as company_id,
+              r.rate,
+              r.name AS date_start,
+              (SELECT name FROM res_currency_rate r2
+              WHERE r2.name > r.name AND
+                    r2.currency_id = r.currency_id AND
+                    (r2.company_id is null or r2.company_id = c.id)
+               ORDER BY r2.name ASC
+               LIMIT 1) AS date_end
+            FROM res_currency_rate r
+              JOIN res_company c ON (r.company_id is null or r.company_id = c.id)
+        )
+
+        /* ALL CONFIRMED PURCHASES */
```

Comment:
I think that you might use recursive sql. Here is a link on how to do it...
https://dba.stackexchange.com/questions/311189/child-to-parent-in-same-table-there-must-be-a-better-way-to-do-this/311190#311190